### ISSUE TITLE
FixedEffectModels.jl not handling more than one fixed effect

### ISSUE THREAD

**Issue by chriscandelaria** [FixedEffects/FixedEffectModels.jl#2]
We're receiving the following error when absorbing more than one fixed effect:

```
ERROR: PosDefException(1)
in cholfact! at linalg/factorization.jl:38
in cholfact! at linalg/factorization.jl:32
in reg at /user/.julia/v0.3/FixedEffectModels/src/reg.jl:165
in reg at /user/.julia/v0.3/FixedEffectModels/src/reg.jl:15
```

To replicate this error, we use the following Stata dataset: 

```
use http://www.stata-press.com/data/r9/nlswork.dta
export delim nlswork.csv
```

Sample Julia code is below. Note that a single fixed effect model works fine. You will see that in modout1 and modout2. However, the combination of fixed effects produces the error.

```
using SparseFactorModels, DataFrames, FixedEffectModels
df = readtable("nlswork.csv")
df[:pidcode] = pool(df[:idcode])
df[:pyear] = pool(df[:year])

testfactor = SparseFactorModel(:pidcode, :pyear, 1)

modout1 = reg(ln_wage ~ ttl_exp |> pidcode, df)
modout2 = reg(ln_wage ~ ttl_exp |> pyear, df)
modout3 = reg(ln_wage ~ ttl_exp |> pidcode + pyear, df)
```

**Comment by matthieugomez** [comment]
I just tried and can't reproduce the error. `modout3` returns the same coefficients than `reghdfe` in Stata. This error basically means that the matrix `X'X` is not invertible (where X is the demeaned variable `ttl_exp`)

**Comment by matthieugomez** [comment]
Solved by updating to 0.4
